src/category_theory/monad/algebra.lean

```diff
@@ -0,0 +1,111 @@
+/-
+Copyright (c) 2019 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+import category_theory.monad.basic
+import category_theory.adjunction.basic
+
+namespace category_theory
+open category
+
+universes v₁ u₁ -- declare the `v`'s first; see `category_theory.category` for an explanation
+
+variables {C : Type u₁} [𝒞 : category.{v₁} C]
+include 𝒞
+
+namespace monad
+
+structure algebra (T : C ⥤ C) [monad.{v₁} T] : Type (max u₁ v₁) :=
+(A : C)
+(a : T.obj A ⟶ A)
+(unit' : (η_ T).app A ≫ a = 𝟙 A . obviously)
+(assoc' : ((μ_ T).app A ≫ a) = (T.map a ≫ a) . obviously)
+
+restate_axiom algebra.unit'
+restate_axiom algebra.assoc'
+
+namespace algebra
+variables {T : C ⥤ C} [monad.{v₁} T]
+
```

Comment:
I think we still haven't figured out what is best... but from the point of readability it would be very nice to set up coercions for `algebra.A` and `hom.f`. Also... wouldn't `algebra.carrier` be more in line with other "bundlings"? Like `submodule.carrier` etc...? Especially if we don't set up coercions, I would prefer to stick to a predictable naming scheme.